In [1]:
import pandas

# Exact String Match

In [7]:
df = pandas.read_pickle("results/NMVWtoWikiExactMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

3360

In [8]:
df.loc[df["match"] == "NO"].sample(20)

nmvw_uri  \
5656  https://hdl.handle.net/20.500.11840/pi80338   
4495  https://hdl.handle.net/20.500.11840/pi69977   
4666  https://hdl.handle.net/20.500.11840/pi76699   
280   https://hdl.handle.net/20.500.11840/pi22767   
5436  https://hdl.handle.net/20.500.11840/pi79507   
94    https://hdl.handle.net/20.500.11840/pi20636   
886   https://hdl.handle.net/20.500.11840/pi40058   
316   https://hdl.handle.net/20.500.11840/pi24161   
1917   https://hdl.handle.net/20.500.11840/pi4758   
5139   https://hdl.handle.net/20.500.11840/pi7817   
1425  https://hdl.handle.net/20.500.11840/pi44597   
5455  https://hdl.handle.net/20.500.11840/pi79693   
3205  https://hdl.handle.net/20.500.11840/pi63722   
1134  https://hdl.handle.net/20.500.11840/pi43665   
997   https://hdl.handle.net/20.500.11840/pi40694   
1210  https://hdl.handle.net/20.500.11840/pi43969   
1486  https://hdl.handle.net/20.500.11840/pi44862   
5438  https://hdl.handle.net/20.500.11840/pi79589   
5309  https://hdl.handle.net/20.500.11840/pi78862   
2323  https://hdl.handle.net/20.500.11840/pi60066   

                                    name_label  \
5656                  A.B. (Ada Bird) Petyarre   
4495                                Shi Zichen   
4666                                Albert, R.   
280      J.C. (Jacob Christiaan) Koningsberger   
5436                              O. Zerkowitz   
94        J.S.G. (Jan Simon Gerardus) Gramberg   
886                           dhr. J. Lie a Fo   
316                             C. (Carl) Bock   
1917       Prof. Dr. J.A. (Jan Antonie) Honing   
5139                            Dr. A.B. Meyer   
1425                       W. (Wilfrid) Daleus   
5455                          Utagawa Kunimori   
3205                               L. Scherman   
1134                               Jr. Goemans   
997                        Fam.  van den Akker   
1210            J. (John) Cleveley the Younger   
1486  Prof. J.F. (Johann Friedrich) Blumenbach   
5438                             Böeseken, A.   
5309                               Ephraim, J.   
2323                            Maki Kinnosuke   

                                      wiki_uri  \
5656  http://www.wikidata.org/entity/Q27077624   
4495  http://www.wikidata.org/entity/Q82575085   
4666  http://www.wikidata.org/entity/Q83286830   
280    http://www.wikidata.org/entity/Q2292859   
5436  http://www.wikidata.org/entity/Q83290955   
94    http://www.wikidata.org/entity/Q16062716   
886    http://www.wikidata.org/entity/Q3929328   
316    http://www.wikidata.org/entity/Q1036746   
1917  http://www.wikidata.org/entity/Q63003525   
5139     http://www.wikidata.org/entity/Q63167   
1425  http://www.wikidata.org/entity/Q51048361   
5455  http://www.wikidata.org/entity/Q56615009   
3205    http://www.wikidata.org/entity/Q214847   
1134  http://www.wikidata.org/entity/Q86818241   
997   http://www.wikidata.org/entity/Q86818159   
1210   http://www.wikidata.org/entity/Q6226371   
1486     http://www.wikidata.org/entity/Q58588   
5438  http://www.wikidata.org/entity/Q60233595   
5309  http://www.wikidata.org/entity/Q18608010   
2323  http://www.wikidata.org/entity/Q86818816   

                                             wiki_label retrieved_names match  
5656  [Ada Petyarre Bird, Ada Bird Petyarre, Ada Pet...              []    NO  
4495               [Zichen Shi, Zichen Shi, Zichen Shi]              []    NO  
4666  [R. Albert, R. Albert, R. Albert, R. Albert, R...              []    NO  
280   [Jacob Christiaan Koningsberger, Jacob Christi...              []    NO  
5436                                                 []              []    NO  
94    [Jan Simon Gerardus Gramberg, Jan Simon Gerard...              []    NO  
886   [John Lie A Fo, John Lie A Fo, John Lie A Fo, ...              []    NO  
316   [Carl Bock, Carl Bock, كارل بوك, كارل بوك, Car...              []    NO  
1917  [Jan Antonie Honing, Jan Antonie Honing, Jan A...              []    NO  
5139  [أدولف ماير, ادولف 

Reason for Mismatch: 
    - Different data keeping practices, it varies depending on who kept the record A.B. (Ada Bird) Petyarre	--> Ada Petyarre Bird; both initial and name in the bracket is present
    - position mismatch for firstname and lastname: Shi Zichen --> Zichen Shi
    - Surname first then abbreviation: Albert, R. --> R. Albert
    - Partial abbreviation: J. Lie a Fo --> John Lie A Fo
    - addressed with Dhr. or Mw or mevr
    - Title, e.g., Prof. Dr. present: Prof. Dr. J.A. (Jan Antonie) Honing --> Jan Antonie Honing
    - The name is mention through abbreviation, not with full name: L. Scherman --> Lucian Scherman
    - Mentioned with just middle name: W. (Wilfrid) Daleus --> Joseph Wilfrid Daleus
    - character encoding went wrong: Böeseken, A.--> A. Böeseken
    - why 997 is not a match
    - why 2323 and 5455 is empty
    - The word junior is mentioned in many different way: Jr. Goemans --> Goemans Jr, J. (John) Cleveley the Younger --> John, Jr. Cleveley, John, Jr. Cleveley,...

# Surname Matching

In [10]:
df = pandas.read_pickle("results/NMVWtoWikiSurnameMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

5245

In [12]:
df.loc[df["match"] == "NO"].sample(20)

nmvw_uri  \
930   https://hdl.handle.net/20.500.11840/pi40328   
2424  https://hdl.handle.net/20.500.11840/pi60258   
2389  https://hdl.handle.net/20.500.11840/pi60197   
4950  https://hdl.handle.net/20.500.11840/pi77587   
561   https://hdl.handle.net/20.500.11840/pi38705   
5364  https://hdl.handle.net/20.500.11840/pi79059   
5262  https://hdl.handle.net/20.500.11840/pi78602   
4071  https://hdl.handle.net/20.500.11840/pi66980   
3272  https://hdl.handle.net/20.500.11840/pi63913   
2493  https://hdl.handle.net/20.500.11840/pi60416   
1154  https://hdl.handle.net/20.500.11840/pi43692   
4875  https://hdl.handle.net/20.500.11840/pi77208   
2622  https://hdl.handle.net/20.500.11840/pi61329   
4732  https://hdl.handle.net/20.500.11840/pi76836   
559   https://hdl.handle.net/20.500.11840/pi38703   
4784  https://hdl.handle.net/20.500.11840/pi76939   
6088   https://hdl.handle.net/20.500.11840/pi8190   
2059  https://hdl.handle.net/20.500.11840/pi57990   
1469  https://hdl.handle.net/20.500.11840/pi44781   
5347  https://hdl.handle.net/20.500.11840/pi79018   

                                         name_label  \
930                               M. (Metin) Özlen   
2424                            Shunkôsai Hokushû   
2389                                    Seki Yûshi   
4950                                      G. Wigand   
561                                     Z. Prévost   
5364                                   Sutton, F.W.   
5262                                   Warden, D.B.   
4071                        Uoya Eikichi (uitgever)   
3272                    Tjokorda Gdé Raka Sukewati   
2493  Watei, Ryuho, ...., Seiko, Hazan & (yosegaki)   
1154                         Anak Agung Gedeh Mereg   
4875                            Valentyn V.D.M., F.   
2622                       Kobori Sôjitsu Masaharu   
4732                                   Harneman, W.   
559                              Friedrich Salathé   
4784                                  Ondei, G.M.D.   
6088                                          Turan   
2059                    C.W. (Carl Wilhelm) Lüders   
1469                          Nynke Jelles-Schepers   
5347                                     Hughan, A.   

                                      wiki_uri  \
930   http://www.wikidata.org/entity/Q82574219   
2424   http://www.wikidata.org/entity/Q3074677   
2389  http://www.wikidata.org/entity/Q86818944   
4950  http://www.wikidata.org/entity/Q84594433   
561   http://www.wikidata.org/entity/Q50810310   
5364  http://www.wikidata.org/entity/Q29043149   
5262  http://www.wikidata.org/entity/Q18122726   
4071  http://www.wikidata.org/entity/Q84498310   
3272  http://www.wikidata.org/entity/Q86819769   
2493  http://www.wikidata.org/entity/Q86819124   
1154  http://www.wikidata.org/entity/Q86818249   
4875    http://www.wikidata.org/entity/Q980045   
2622  http://www.wikidata.org/entity/Q84498630   
4732  http://www.wikidata.org/entity/Q83286786   
559    http://www.wikidata.org/entity/Q3087757   
4784  http://www.wikidata.org/entity/Q83286463   
6088  http://www.wikidata.org/entity/Q86817975   
2059  http://www.wikidata.org/entity/Q18018957   
1469  http://www.wikidata.org/entity/Q17300238   
5347  http://www.wikidata.org/entity/Q52218640   

                                             wiki_label retrieved_names match  
930             [Metin Özlen, Metin Özlen, Metin Özlen]              []    NO  
2424  [شونكوساى هوكوشو, Shunkōsai Hokushū, Shunkōsai...              []    NO  
2389                           [Seki Yûshi, Seki Yûshi]              []    NO  
4950                                                 []              []    NO  
561   [Zachée Prévost, Zachée Prévost, Zachée Prévos...              []    NO  
5364  [Frederick William Sutton, Frederick William S...              []    NO  
5262  [David Baillie Warden, David Baillie Warden, D...              []    NO  
4071  [Uoya Eikichi, Uoya Eikichi, 魚屋栄吉, Uoya Eikich...              []    NO  


Reason for Mismatch: 
1. Spelling variation of surname: M. (Metin) Ãzlen --> Metin Özlen; Valentyn --> Valentijn
2.  Character encoding went wrong: Shunkôsai Hokushû --> Shunkōsai Hokushū; Seki Yûshi --> Seki Yûshi; Prévost --> Prévost
3. Difficulties in finding surname:
    a. Surname went first like author reference: Warden, D.B. --> David Baillie Warden
    b. not necessary the last word of the name field is surname, sometimes it specifies smth about the person. Anak Agung Gedeh Mereg --> Mereg Anak Agung Gedeh; 
    c. Valentyn V.D.M., F. --> François Valentijn; 
    d. Kobori Sôjitsu Masaharu --> Sôjitsu Masaharu Kobori
4. Group-of people is also considered as instance of human: Watei, Ryuho, ...., Seiko, Hazan & (yosegaki) --> Watei, Ryuho, ...., Seiko, Hazan & (yosegaki) SPL 

- why 4950 and 3272  is empty for wiki_label

# Abbreviation Matching

In [13]:
df = pandas.read_pickle("results/NMVWtoWikiAbbreviationMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

3662

In [14]:
df.loc[df["match"] == "NO"].sample(20)

nmvw_uri  \
5111  https://hdl.handle.net/20.500.11840/pi78041   
4494  https://hdl.handle.net/20.500.11840/pi69976   
4538   https://hdl.handle.net/20.500.11840/pi7118   
2347   https://hdl.handle.net/20.500.11840/pi6011   
1307  https://hdl.handle.net/20.500.11840/pi44178   
5751  https://hdl.handle.net/20.500.11840/pi80591   
4317  https://hdl.handle.net/20.500.11840/pi69372   
6097  https://hdl.handle.net/20.500.11840/pi81917   
5389  https://hdl.handle.net/20.500.11840/pi79303   
955   https://hdl.handle.net/20.500.11840/pi40441   
4369  https://hdl.handle.net/20.500.11840/pi69601   
4952  https://hdl.handle.net/20.500.11840/pi77594   
1635  https://hdl.handle.net/20.500.11840/pi45748   
1651  https://hdl.handle.net/20.500.11840/pi45807   
447    https://hdl.handle.net/20.500.11840/pi3628   
6098  https://hdl.handle.net/20.500.11840/pi81918   
4903  https://hdl.handle.net/20.500.11840/pi77407   
2104   https://hdl.handle.net/20.500.11840/pi5888   
21     https://hdl.handle.net/20.500.11840/pi1369   
2355  https://hdl.handle.net/20.500.11840/pi60126   

                                  name_label  \
5111                        Sadoon, S.D.H.M.   
4494                           Wei Changqing   
4538                    H. (Herman) Stafhell   
2347        C. (Christiaan) Snouck Hurgronje   
1307                  Dhr. K. (Kommer) Klein   
5751                 lgnaz Sebastian Klauber   
4317    G.F.B. (George Felix Bernard) Watrin   
6097      G.M. (Gustaaf Martinus) Oosterling   
5389                             Brugsch, E.   
955                       H. (Hamid) Sharifi   
4369                          I Ketut Tungeh   
4952                           A. van Tricht   
1635                    C. (Carlos) Amorales   
1651                   M. (Marina) Abramovic   
447              Dhr. M. (Mathijs) de Vreede   
6098        N.W. (Nicolaas Wilhelm) Jungmann   
4903                           Hellfarth, C.   
2104  A.P.H. (Albertus Paulus Hermanus) Hotz   
21         H.D. (Hieronijmus David) Fraenkel   
2355                             Niwa Tôkei   

                                      wiki_uri  \
5111  http://www.wikidata.org/entity/Q83290981   
4494  http://www.wikidata.org/entity/Q84499742   
4538     https://rkd.nl/explore/artists/385437   
2347    http://www.wikidata.org/entity/Q731829   
1307  http://www.wikidata.org/entity/Q86736196   
5751  http://www.wikidata.org/entity/Q38073999   
4317  http://www.wikidata.org/entity/Q82576123   
6097  http://www.wikidata.org/entity/Q64969957   
5389     http://www.wikidata.org/entity/Q73057   
955   http://www.wikidata.org/entity/Q82575034   
4369  http://www.wikidata.org/entity/Q82575898   
4952  http://www.wikidata.org/entity/Q86741453   
1635   http://www.wikidata.org/entity/Q5041755   
1651     http://www.wikidata.org/entity/Q47496   
447   http://www.wikidata.org/entity/Q86735130   
6098   http://www.wikidata.org/entity/Q7028792   
4903  http://www.wikidata.org/entity/Q86741387   
2104   http://www.wikidata.org/entity/Q2639541   
21    http://www.wikidata.org/entity/Q60238568   
2355  http://www.wikidata.org/entity/Q11368639   

                                             wiki_label        abbreviations  \
5111  [S.D.H.M. Sadoon, S.D.H.M. Sadoon, S.D.H.M. Sa...          S. S.D.H.M.   
4494  [Changqing Wei, Changqing Wei, Changqing Wei, ...         W. Changqing   
4538                                                 []       H. (. Stafhell   
2347  [Snouck Hurgronje, Snouck Hurgronje, Snouck Hu...   C. (. S. Hurgronje   
1307                                                 []       D. K. (. Klein   
5751  [Ignaz Sebastian Klauber, Ignaz Sebastian Klau...        l. S. Klauber   
4317  [George Felix Bernard Watrin, George Felix Ber...   G. (. F. B. Watrin   
6097  [جوستاف مارتينوس اوستيرلينج, Gustaaf Martinus ...  G. (. M. Oosterling   
5389  [اميل بروجستش, Émile Charles Albert Brugsch, E...                B. E.   
955       [Hamid Sharifi, Hamid Sharifi, Hamid Sharifi]        H. (

Reason for Mismatch: 
- Not surname in the last: Sadoon, S.D.H.M. --> S.D.H.M. Sadoon --> S. S.D.H.M.(abbrv.)	
    - Wei Changqing --> Changqing Wei --> Changqing Wei
    - Brugsch, E. --> B. E.
- there is often mention like this: H. (Herman) Stafhell; while abbreviated it went like this,  H. (. Stafhell; A good portion of mismatch happended due to this
- Character encoding went wrong: Niwa Tôkei--> Niwa Tōkei --> N. Tôkei(abbrv.)	

# Fuzzy String Matching

In [15]:
df = pandas.read_pickle("results/NMVWtoWikiFuzzyStringMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

6120

In [16]:
df.loc[df["match"] == "NO"].sample(20)

nmvw_uri                 name_label  \
1283  https://hdl.handle.net/20.500.11840/pi44123   Dhr. S. (Serge) Debecque   
2245  https://hdl.handle.net/20.500.11840/pi59907              Jôyô Tôkai   
5192  https://hdl.handle.net/20.500.11840/pi78377       Prof. Dr. K. Florenz   
3513  https://hdl.handle.net/20.500.11840/pi64838                   V. Mutwa   
457   https://hdl.handle.net/20.500.11840/pi37979               W.F. Winckel   
2125  https://hdl.handle.net/20.500.11840/pi59268                P.W.M. Trap   
2013  https://hdl.handle.net/20.500.11840/pi57056         prof E.H. Giglioli   
4854  https://hdl.handle.net/20.500.11840/pi77121              Solvyns, F.B.   
3251  https://hdl.handle.net/20.500.11840/pi63849     jhr mr J.E. de Sturler   
535   https://hdl.handle.net/20.500.11840/pi38363        S. (Sıtkı) Olçar   
5951  https://hdl.handle.net/20.500.11840/pi81521                  Gu Kaizhi   
3314  https://hdl.handle.net/20.500.11840/pi64047                H.J. Wehrli   
4924  https://hdl.handle.net/20.500.11840/pi77520                  G. T. Bom   
1454  https://hdl.handle.net/20.500.11840/pi44754        dhr. O. (Obed) Addo   
4785  https://hdl.handle.net/20.500.11840/pi76944               Penn, A.T.W.   
3503  https://hdl.handle.net/20.500.11840/pi64727             Teodora Blanco   
3035  https://hdl.handle.net/20.500.11840/pi63191                C.S. Kröge   
1265  https://hdl.handle.net/20.500.11840/pi44065  Dhr. G. (Georges) Delerue   
3026  https://hdl.handle.net/20.500.11840/pi63148                  A. Koëhn   
1275  https://hdl.handle.net/20.500.11840/pi44083  Dhr. A. (Alphons) Hustinx   

                                      wiki_uri  \
1283  http://www.wikidata.org/entity/Q27064216   
2245  http://www.wikidata.org/entity/Q86818680   
5192  http://www.wikidata.org/entity/Q86741784   
3513  http://www.wikidata.org/entity/Q84594545   
457   http://www.wikidata.org/entity/Q82576026   
2125  http://www.wikidata.org/entity/Q30304094   
2013    http://www.wikidata.org/entity/Q692160   
4854   http://www.wikidata.org/entity/Q2682432   
3251  http://www.wikidata.org/entity/Q86739702   
535   http://www.wikidata.org/entity/Q82574190   
5951  http://www.wikidata.org/entity/Q84498394   
3314  http://www.wikidata.org/entity/Q86739884   
4924  http://www.wikidata.org/entity/Q18654981   
1454  http://www.wikidata.org/entity/Q84593123   
4785  http://www.wikidata.org/entity/Q52218753   
3503  http://www.wikidata.org/entity/Q18163000   
3035  http://www.wikidata.org/entity/Q86739103   
1265    http://www.wikidata.org/entity/Q435034   
3026  http://www.wikidata.org/entity/Q86739107   
1275  http://www.wikidata.org/entity/Q55435497   

                                             wiki_label retrieved_names match  
1283  [Serge Debecque, Serge Debecque, Serge Debecqu...              []    NO  
2245                           [Jôyô Tôkai, Jôyô Tôkai]              []    NO  
5192                                                 []              []    NO  
3513                                                 []              []    NO  
457   [Willem Ferdinand Winckel, Willem Ferdinand Wi...              []    NO  
2125  [Pieter Willem Marinus Trap, Pieter Willem Mar...              []    NO  
2013  [Amphiptera, انريكو هيلير جيجليولى, Amphiptera...              []    NO  
4854  [بالثازار سولڤينس, Balthazar Solvyns, Balthaza...              []    NO  
3251  [J.E. de Sturler, Jacques Edouard de Sturler, ...              []    NO  
535             [Sıtkı Olçar, Sıtkı Olçar, Sıtkı Olcar]              []    NO  
5951                                                 []              []    NO  
3314                                                 []              []    NO  
4924  [Gerardus Theodorus Bom, Gerardus Theodorus Bo...              []    NO  
1454                             [Obed Addo, Obed Addo]              []    NO  
4785  [Albert Thomas Watson Penn, Albert Thomas Wats...              []    NO  
3503  [تيودورا بلانكو نونيز, 

Maybe, a good idea would be to write another code that looks for surname in the whole candiate name. i.e,
- row['LastName'] in candidate.split()

Maybe, it is a good idea to try to seperate title like, Dhr. Mevr. Prof. Dr. ...

Maybe, it is a good idea to detect surname automatically somehow using AI. 

Maybe, we can remove everuthing that falls in bracket (); causing a lot of mismatch for abbreviation match